admin/js/profiles.livesession.js

```diff
@@ -18,72 +18,105 @@
+        $('#spice-screen').html('');
+        connecting = null;
+        noretry = true;
+        console.log('FC: Connection tries timed out');
+        $('#spinner-modal').modal('hide');
+        showMessageDialog('Connection error to virtual machine.', 'Connection error');
+      }, conn_timeout);
+    }
+  }
 
-function spiceClientConnection(host, port) {
+  function spice_connected() {
+    console.log('FC: Connected to virtual machine using SPICE');
+    $('#spinner-modal').modal('hide');
+    if (connecting) {
+      clearTimeout(connecting);
+      connecting = null;
+    }
+  }
 
   function spice_error(err) {
+    console.log("FC: SPICE connection error:", err.message);
+
+    set_connection_timeout()
 
-    console.error(err);
-    // Try to reconnect
-    if (!retrying && sc.state != 'ready') {
-      retrying = setTimeout(function() {
-        console.log('Trying to reconnect');
+    if (err.message == 'Unexpected close while ready' || err.message == 'Connection timed out.' || sc.state != 'ready')  {
+      if (!noretry) {
+        $('#spinner-modal h4').text('Connecting to virtual machine. Please wait...');
+        $('#spinner-modal').modal('show');
         do_connection();
-        retrying = null;
-      }, 200);
+      }
+    } else {
+      $('#spinner-modal').modal('hide');
+      if (connecting) {
+        clearTimeout(connecting);
+        connecting = null;
+      }
+      showMessageDialog('Connection error to virtual machine.', 'Connection error');
     }
-  }
 
-  function agent_connected(sc) {
-    window.addEventListener('resize', handle_resize);
-    window.spice_connection = this;
-    resize_helper(this);
-    if (window.File && window.FileReader && window.FileList && window.Blob) {
-      var spice_xfer_area = document.createElement("div");
-      spice_xfer_area.setAttribute('id', 'spice-xfer-area');
-      document.getElementById('spice-area').appendChild(spice_xfer_area);
-      document.getElementById('spice-area').addEventListener('dragover', handle_file_dragover, false);
-      document.getElementById('spice-area').addEventListener('drop', handle_file_drop, false);
-    }
-    else {
-      console.log("File API is not supported");
-    }
   }
 
   function do_connection() {
-    console.log('Connecting to spice session')
+    console.log('FC: Connecting to spice session')
+    if (sc) sc.stop()
+    $('#spice-screen').html('');
     sc = new SpiceMainConn({
       uri: 'ws://' + location.hostname + ':' + port,
-      screen_id: "spice-screen",
+      screen_id: 'spice-screen',
+      onsuccess: spice_connected,
+      onerror: spice_error
       // dump_id: "debug-div",
       // message_id: "message-div",
       // password: password,
-      // onagent: agent_connected
-      onerror: spice_error
+      // onagent: agent_connected,
     });
   }
 
   try {
     do_connection();
   } catch (e) {
-    console.error('Fatal error:' + e.toString());
+    console.error('FC: Fatal error:' + e.toString());
```

Comment:
Because in console log, there are a lot of messages with a similar structure like Error, Fatal Error etc. And Sometimes I'm not able to know in a look if the message is something we are writing or is the SPICE client.

For that I prepended all messages using FC: